docs/proposals/scheduling/average-replicas-assign/average-replicas-assign.md

```diff
@@ -0,0 +1,275 @@
+
+<!--
+因此，Karmada 引入这项改动是能够兼容过往用户的配置的，该 Proposal 目前不存在任何可预见性的风险。
+-->
+
+## Design Details
+
+### API Design
+
+In Karmada, the API related to scheduling is the PropagationPolicy. Currently, its API design is as follows:
+
+<!--
+在 Karmada 中，和调度相关的 API 即为 PropagationPolicy，目前，其 API 设计如下：
+-->
+
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: PropagationPolicy 
+metadata:
+spec:
+  placement:
+    replicaScheduling: 
+      replicaDivisionPreference: Average
+```
+
+As we can see, to add an AverageReplicas replica allocation strategy, we can consider adding a new value, "Average," to the replicaDivisionPreference. Here is an example.
+
+<!--
+可以看到，如果要添加一个 AverageReplicas 副本分配策略，可以考虑给 replicaDivisionPreference 添加一个新的取值 Average。下面是一个例子：
+-->
+
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: PropagationPolicy
+metadata:
+  name: foo
+  namespace: bar
+spec:
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: foo
+  placement:
+    replicaScheduling:
+      replicaSchedulingType: Divided
+      replicaDivisionPreference: Average
+```
+
+Note that `replicaDivisionPreference == Average` will only be effective when `replicaSchedulingType == Divided`.
+
+<!--
+注意，replicaDivisionPreference == Average 只有在 replicaSchedulingType == Divided 时才会生效。
+-->
+
+### Process Design
+
+Currently, Karmada's scheduling process consists of four phases: Filter, Score, Select, and Assign. To achieve an even distribution of replicas across the clusters as much as possible, it is necessary to implement the allocation of replicas to each member cluster in an as-even-as-possible manner during the Assign phase.
+
+<!--
+目前 Karmada 的调度分为四个阶段 Filter、Score、Select、Assign。为了能够尽可能地满足将副本平均分配到各个集群上，需要在 Assign 阶段，实现对各个工作集群进行尽力平均地副本分配。
+-->
+
+#### Assign
+
+We need to add the AverageReplicas replica allocation strategy in the Assign phase. Since the cluster selection has already been completed in the Select phase, the input for the Assign phase will be:
+
+1. The selected clusters.
+2. The number of replicas to be allocated.
+3. ResourceBindingSpec, which includes the information of the new replica allocation strategy.
+
+<!--
+我们需要在 Assign 阶段添加副本分配策略 AverageReplicas。由于已经在 Select 阶段完成了Cluster 的筛选，因此我们在 Assign 阶段的输入即是：
+1.被选出的 Cluster。
+2.待分配的 Replica（个数）。
+3.ResourceBindingSpec，其中包括新增的副本数分配策略信息。
+-->
+
+The logic here is essentially to determine how many replicas should be allocated to each cluster. According to the project's objectives, in the Assign phase, replicas should be evenly distributed among the selected clusters as much as possible. First, we need to calculate the ideal number of replicas each cluster should be allocated:
+
+1. Define the number of selected clusters as L。
+2. Calculate the average number of replicas each cluster should be allocated, defined as avg_rep = Replica / L。
```

Comment:
How to solve the non-divisible scenario, like `L=3` and `Replica=8`, then the `avg_rep = 2`, follow the algorithm metioned below, only `avg_rep * L = 6` may be allocated.
I think we should set the `sumPendingReplica=Replica - L * avg_rep = 2`.

Reply:
I have refined the explanation of this issue, thanks.